source/common/upstream/cluster_manager_impl.cc

```diff
@@ -729,9 +736,10 @@ bool ClusterManagerImpl::removeCluster(const std::string& cluster_name) {
   return removed;
 }
 
-void ClusterManagerImpl::loadCluster(const envoy::config::cluster::v3::Cluster& cluster,
-                                     const std::string& version_info, bool added_via_api,
-                                     ClusterMap& cluster_map) {
+absl::optional<ClusterManagerImpl::ClusterDataPtr>
```

Comment:
nit: I don't think this needs to be an optional, just return a pointer or not.

Reply:
Right. I updated the signature and the comment